src/command.rs

```diff
@@ -518,6 +520,94 @@ pub struct Credential<'l> {
     pub password_safe: Option<PasswordSafeData<'l>>,
 }
 
+#[derive(Clone, Copy, Eq, PartialEq, Debug, Default)]
+pub struct CredentialUpdate<'l> {
+    pub label: &'l [u8],
+    pub new_label: Option<&'l [u8]>,
+    pub properties: Option<Properties>,
+    pub password_safe: Option<PasswordSafeData<'l>>,
+}
+
+impl<'l, const C: usize> TryFrom<&'l Data<C>> for CredentialUpdate<'l> {
+    type Error = Status;
+    fn try_from(data: &'l Data<C>) -> Result<Self, Self::Error> {
+        use flexiber::TaggedSlice;
+        let mut decoder = flexiber::Decoder::new(data);
+
+        let first: TaggedSlice = decoder.decode().map_err(|_| FAILED_PARSING_ERROR)?;
+        ensure(
+            first.tag() == (Tag::Name as u8).try_into().unwrap(),
+            FAILED_PARSING_ERROR,
+        )?;
+        let label = first.as_bytes();
+
+        if label.is_empty() {
+            return Err(FAILED_PARSING_ERROR);
+        }
+        // end of obligatory fields parsing
+
+        let mut res = CredentialUpdate {
+            label,
+            ..Default::default()
+        };
+
+        // TODO This would be better modeled with Option, than empty slice,
+        //  to allow fields removal by setting the request fields to empty slice.
+        //  Currently only setting is possible. A workaround for now is to
+        //  set the value to remove to " ".
+        let mut pws = PasswordSafeData {
+            login: &[],
+            password: &[],
+            metadata: &[],
```

Comment:
Why not do it now with an option?
I feel like it's better to do it the right way from the start.

Also with an option `None` could mean "unchanged"  (as I would expect most updates to only change one field).

Reply:
I would actually do it with:

- `None` means unchanged
- `Some([])` means delete (i think we may want to reject empty slices also at registration
- `Some([..])` means update this part.

This feels more intuitive (to me), and is consistent with how deletion works in PIV for example.

Reply:
Yup, this sounds good. I've intended to this this way (as in comment).
This looked like a bigger refactor (as you've mentioned, involves register command), hence I wanted to extract this to a separate task to keep it clear and focused.

Reply:
I don't think it is big enough to justify be done in another PR, but why not.

I still think we need to hold merging https://github.com/Nitrokey/pynitrokey/pull/425#pullrequestreview-1576442445 in that case before the refactor is done.

Reply:
Updated as mentioned. Resolving.